File: turnstile-backend/src/main/java/it/tdgc/turnstile/dto/UserDTO.java
package it.tdgc.turnstile.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;


@Getter
@Setter
@Component
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {
    private String usertype;
    private String name;
    private String surname;
    private String email;
    private RoleDTO role;
    private CompanyDTO company;
    private BadgeDTO badge;
}

File: turnstile-frontend/src/app/core/dto/user-dto.ts
import {RoleDTO} from './role-dto';
import {CompanyDTO} from './company-dto';
import {BadgeDTO} from './badge-dto';

export class UserDTO {
    usertype = "";
    name= "";
    surname= "";
    email= "";
    role = new RoleDTO();
    company = new CompanyDTO();
    badge = new BadgeDTO();

    constructor(init?: Partial<UserDTO>) {
        Object.assign(this, init);
      }
}
